cpp/include/kvikio/shim/libcurl.hpp

```diff
@@ -249,12 +250,31 @@ class CurlHandle {
   }
 };
 
+namespace detail {
+/**
+ * @brief Fix Conda's manipulation of __FILE__.
+ *
+ * Conda manipulate the path information in its shared libraries with the results that the C macro
+ * `__FILE__` might contain trailing `\0` chars. Normally, this isn't a problem because `__FILE__`
+ * is a `const char*` that are terminated by the first encounter of `\0`. However, when creating a
+ * `std::string` from a `char*`, the compiler might optimize the code such that the `std::string`
+ * is created from the full size of `__FILE__` including the trailing `\0` chars.
+ */
+__attribute__((optnone)) inline std::string fix_conda_file_path_hack(std::string filename)
```

Comment:
*issue*: I think we should explicitly request `noinline` rather than hoping that `optnone` will ensure the function is not inlined.


```suggestion
__attribute__((noinline)) inline std::string fix_conda_file_path_hack(std::string filename)
```

Reply:
I think we can do both, no?

```
__attribute__((optnone)) __attribute__((noinline)) std::string fix_conda_file_path_hack(std::string filename)
```

Reply:
Interesting, somehow it's not in GCC's but it's in Clang's saying it's available in GNU (I assume GCC): https://clang.llvm.org/docs/AttributeReference.html#optnone

Reply:
Maybe it should be `__attribute__((optimize(0)))` instead?

Reply:
> Maybe it should be `__attribute__((optimize(0)))` instead?

Does clang support `optimize(0)` ?

Reply:
`optimize` is undocumented in Clang. 🤣 

I guess you'll either have to risk leaving `optnone` or do some compiler checking.

Reply:
Heh, lovely :)

Reply:
Let's see if `__attribute__((noinline))` is enough